src/Microsoft.Identity.Web/WebApiAuthenticationBuilderExtensions.cs

```diff
@@ -74,6 +74,7 @@ public static class WebApiAuthenticationBuilderExtensions
 
             builder.Services.AddHttpContextAccessor();
             builder.Services.AddSingleton<IJwtBearerMiddlewareDiagnostics, JwtBearerMiddlewareDiagnostics>();
+            builder.Services.AddHttpClient();
```

Comment:
Just for my understanding - this will ensure that the high performance HTTP client factory is used? Is there any scenario where people might not want to use this, for example for testing?

Reply:
That's a good question. 
Would it be ok to wait for customer feedback ?

Reply:
Absolutely.

Reply:
[AddHttpClient()](https://github.com/dotnet/runtime/blob/master/src/libraries/Microsoft.Extensions.Http/src/DependencyInjection/HttpClientFactoryServiceCollectionExtensions.cs#L23) adds a few required services. Adding my own factory should still work. I can either remove these default services and add my own before the service provider is built. I can also just add my own factory after I call AddHttpClient, it should overwrite the same service types that were added earlier. AddHttpClient also uses TryAdd* methods to add the services, so if I add my own factory to the collection first, calling AddHttpClient should not overwrite what I've added.

I haven't read the docs, so there might be some behavior that I'm missing; but I think this should work.